core/src/main/java/com/netease/arctic/io/FileNameHandle.java

```diff
@@ -0,0 +1,149 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.io;
+
+import com.netease.arctic.data.DataFileType;
+import com.netease.arctic.data.DataTreeNode;
+import com.netease.arctic.data.DefaultKeyedFile;
+import com.netease.arctic.io.writer.TaskWriterKey;
+import com.netease.arctic.utils.IdGenerator;
+import com.netease.arctic.utils.TableFileUtils;
+import org.apache.iceberg.FileFormat;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class FileNameHandle {
+
+  private static final String KEYED_FILE_NAME_PATTERN_STRING = "(\\d+)-(\\w+)-(\\d+)-(\\d+)-(\\d+)-.*";
+  private static final Pattern KEYED_FILE_NAME_PATTERN = Pattern.compile(KEYED_FILE_NAME_PATTERN_STRING);
+
+  private static final String FORMAT = "%d-%s-%d-%05d-%d-%s-%05d";
+
+  private FileFormat fileFormat;
+  private final int partitionId;
+  private final long taskId;
+  private final long transactionId;
+
+  // uuid avoid duplicated file name
+  private final String operationId = IdGenerator.randomId() + "";
+  private final AtomicLong fileCount = new AtomicLong(0);
+
+  public FileNameHandle(
+      FileFormat fileFormat,
+      int partitionId,
+      Long taskId,
+      Long transactionId) {
+    this.fileFormat = fileFormat;
+    this.partitionId = partitionId;
+    this.taskId = taskId;
+    this.transactionId = transactionId == null ? 0 : transactionId;
+  }
+
+  public String fileName(TaskWriterKey key) {
+    return fileFormat.addExtension(
+        String.format(FORMAT, key.getTreeNode().getId(), key.getFileType().shortName(),
+            transactionId, partitionId, taskId, operationId, fileCount.incrementAndGet()));
+  }
+
+  /**
+   * Flink write transactionId as 0.
+   * if we get transactionId from path is 0, we set transactionId as iceberg sequenceNumber.
+   * @param path file path
+   * @param sequenceNumber iceberg sequenceNumber
+   */
+  public static DefaultKeyedFile.FileMeta parseChange(String path, Long sequenceNumber) {
+    String fileName = TableFileUtils.getFileName(path);
+    Matcher matcher = KEYED_FILE_NAME_PATTERN.matcher(fileName);
+    long nodeId = 1;
+    DataFileType type = null;
+    long transactionId = 0L;
+    if (matcher.matches()) {
+      nodeId = Long.parseLong(matcher.group(1));
+      type = DataFileType.ofShortName(matcher.group(2));
+      transactionId = Long.parseLong(matcher.group(3));
+      transactionId = transactionId == 0 ? sequenceNumber : transactionId;
+    }
+    DataTreeNode node = DataTreeNode.ofId(nodeId);
+    return new DefaultKeyedFile.FileMeta(transactionId, type, node);
+  }
+
+  /**
+   * Path writen by hive can not be pared by arctic format. so we set it transactionId as 0.
+   * @param path file path
+   */
+  public static DefaultKeyedFile.FileMeta parseBase(String path) {
```

Comment:
Could you add more detail about the param path? E.g., path format and demo. I think it will help devs easier to understand what this parsing tries to do.

Reply:
I add some comments for this class to show more details. like 
```
/**
 * File name pattern:${tree_node_id}-${file_type}-${transaction_id}-${partition_id}-${task_id}-{operation_id}-{count}
 * like:
 * 1-B-100-0-0-4217271085623029157-0.parquet
 * 4-ED-101-0-0-9009257362994691056-1.parquet
 * 4-I-101-0-0-9009257362994691056-2.parquet
 */
public class FileNameHandle {
```